tests/extensions/test_pyinstrument.py

```diff
@@ -0,0 +1,49 @@
+import tempfile
+import time
+from pathlib import Path
+
+import strawberry
+from strawberry.extensions import pyinstrument
+
+
+def a():
+    time.sleep(0.1)
+    return b()
+
+
+def b():
+    time.sleep(0.1)
+    return c()
+
+
+def c():
+    time.sleep(0.1)
+    return 4
+
+
+def test_basic_pyinstrument():
+    REPORT_PATH = tempfile.TemporaryFile()
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def the_field(self) -> int:
+            return a()
+
+    schema = strawberry.Schema(
+        query=Query,
+        extensions=[pyinstrument.PyInstrument(report_path=REPORT_PATH)],
+    )
+
+    # Query the schema
+    result = schema.execute_sync("{ theField }")
+
+    assert not result.errors
+    assert result.data["theField"] == 4
+
+    with Path.open(REPORT_PATH, encoding="utf-8") as f:
+        content = f.read()
```

Comment:
similar to above, `Path(REPORT_PATH, encoding="utf-8").read()`

Reply:
https://github.com/strawberry-graphql/strawberry/pull/2727/commits/b43ab659b032cfae5f1a413602fb2799632c2c67